### ISSUE TITLE
Remove the obligation to capitalize `+Page`

### ISSUE THREAD

**Issue by saaymeen** [vikejs/vike#1479]
### Description

This issue is low priority.

When we try to create a page named `+page.tsx`, we are presented with the following error:

> +page.tsx defines an unknown config page, did you mean to define Page instead? (The name of the config Page starts with a capital letter P because it usually defines a UI component: a ubiquitous JavaScript convention is to start the name of UI components with a capital letter.)

While it is true that JSX components must start with a capital letter (**or** contain a dot), the fact that capitalized files are ubiquitous is highly discussable. Yes, it is the preferred approach to file naming by a lot of developers, but React is actually unopinionated about this matter. 

A library like vike should not force a specific convention on users. To show the discrepancy between naming conventions, we can check AirBnB's [style guide](https://github.com/airbnb/javascript/tree/master/react#naming), which is on par with vikes recommendation. When compared to Google's [javascript style guide](https://google.github.io/styleguide/jsguide.html#file-name), we can see that the recommendation here is to use all lowercase letters.

Hand in hand with goes the `+` prepended to a page. With our convention, the .page.tsx extension was much more suitable. 

These points are highly discussable, preferably vike would not force a specific convention to not interfer with conventions of users.

**Comment by brillout** [comment]
I did consider this. But there are other constraints that made me go for the current design. Most notably: redundant syntax adds DX complexity.

Closing because it isn't a priority right now, but I'm happy to revisit if enough people show interest for this.

**Comment by DecentM** [comment]
This creates an issue with cross-platform teams, as case sensitivity works differently across different filesystems. Devs on Mac, Linux, Windows, and WSL can (and will lol) have separate issues when files use capitalisation for development logic differences. The error message says this is a ubiquitous practice, but that's not fully accurate. 

To avoid the extra maintenance burden and silently fix potential issues with this kind of workflow, one option would be to lowercase all filenames before matching. This would technically be a breaking change, because on case sensitive filesystems, devs might have already included files with colliding names post-lowercase. Therefore this solution would need a deprecation warning, and then the update.

I guess in short count that in the interest as well :D

**Comment by brillout** [comment]
> To avoid the extra maintenance burden and silently fix potential issues

What maintenance burden and potential issues are you referring to?

**Comment by DecentM** [comment]
By maintenance burden I'm referring to your earlier comment about redundant syntax adding DX complexity.  

As for potential issues, I vividly remember seeing one particularly nasty one where the `MODULE_NOT_FOUND` error kept cropping up for various people, even if the same commit was fine for others. This usually got resolved with a reboot/re-clone/file-rename-and-back. Especially for devs on Windows. Eventually we adopted a rule where all files must be hyphen-cased, which seemed to solve it.

**Comment by brillout** [comment]
It's all a bit fuzzy; let me know once you can boil it down to a more precise description/reproduction.

**Comment by Zn4rK** [comment]
I don't want to create a new issue for the same thing, but I'm adding this comment to express my interest. I love everything about Vike, except for this. My convention is camelCased, and it bugs me to have files that are capitalised in the project. Could we maybe have it configurable?

**Comment by brillout** [comment]
I ain't categorically against this, but only after we thoroughly discuss and think this through.

> My convention is camelCased

So you don't capitalize the name of your components? That's very unusual, right?

**Comment by Zn4rK** [comment]
> So you don't capitalize the name of your components? That's very unusual, right?

I do, but I prefer to have the file names be camelCased. Sorry, I should've been clearer about it. 

Here's a very simple file structure from a recent react-router project;

<img width="224" height="537" alt="Image" src="https://github.com/user-attachments/assets/11cb28c6-ef53-44e3-8b7f-5dfd7d9d4a54" />

The body component in body.tsx is capitalized and using named exports.
```typescript
export const Body = ({ children }: Props) => {
```

It's just a preference I (and my team(s)) have, and the way I've always structured my code. I don't think it's a deal breaker for me, but it's definitely something that doesn't improve my DX. Since all my other files are also camelCased, seeing "+Page.tsx" in the file tree is a tiny distraction.

**Comment by brillout** [comment]
I see, that's a reasonable request.

I do have one concern left though: I ain't sure whether it's trivial to implement. I don't know how much of Vike's source code assumes the filename to exactly match the setting name. If it's only a small modification to only one function then I'd be up to implement it. One thing you can do is to try to make Vike happy with `+page` by implementing it using your favorite AI (e.g. Augment Code is quite good at these kind of tasks) and see how it goes. Search for the warning you get when you define `+page` instead of `+Page`.
